hoomd/md/CustomForceCompute.h

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2009-2021 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: joaander
+
+#include "hoomd/ForceCompute.h"
+
+#include <map>
+#include <memory>
+
+/*! \file CustomForceCompute.h
+    \brief Declares the backend for computing custom forces in python classes
+*/
+
+#ifdef __HIPCC__
+#error This header cannot be compiled by nvcc
+#endif
+
+#include <pybind11/pybind11.h>
+
+#ifndef __CUSTOMFORCECOMPUTE_H__
+#define __CUSTOMFORCECOMPUTE_H__
+
+namespace hoomd
+    {
+namespace md
+    {
+//! Adds a custom force
+/*! \ingroup computes
+ */
+class PYBIND11_EXPORT CustomForceCompute : public ForceCompute
+    {
+    public:
+    //! Constructs the compute
+    CustomForceCompute(std::shared_ptr<hoomd::SystemDefinition> sysdef);
+
+    //! Destructor
+    ~CustomForceCompute();
+
+    //! Set the python callback
+    void setCallback(pybind11::object py_callback)
+        {
+        m_callback = py_callback;
+        }
+
+    const GlobalArray<Scalar4>& getForceArray() const
+        {
+        return m_force;
+        }
+
+    const GlobalArray<Scalar>& getVirialArray() const
+        {
+        return m_virial;
+        }
+
+    const GlobalArray<Scalar4>& getTorqueArray() const
+        {
+        return m_torque;
+        }
+
+    unsigned int getN() const
+        {
+        return m_pdata->getN();
+        }
+
+    unsigned int getNGhosts() const
+        {
+        return m_pdata->getNGhosts();
+        }
+
+    const GlobalArray<unsigned int>& getRTags() const
+        {
+        return m_pdata->getRTags();
+        }
```

Comment:
Where are these used? Why do they need to be duplicated here instead of calling `m_pdata->method()`?

Reply:
This is more awkwardness to make the templating work. The template class needs to have these methods in order for the `LocalForceComputeData` to compile

Reply:
Exposing the rtags could be removed, as Brandon said, but I think the others need to stay so everything can compile.